src/nfa/nfa.cc

```diff
@@ -472,6 +503,25 @@ State Delta::find_max_state() {
     return max;
 }
 
+std::vector<Post> Delta::transform(const std::function<State(State)>& lambda) const {
+    std::vector<Post> cp_post_vector;
+    cp_post_vector.reserve(num_of_states());
```

Comment:
do we want to rename num_of_states, so that we don't have to search for definition?

Reply:
What do you propose? `size()` is already taken. This is exactly what the value represents. If we name it something else, we will have to search for definition to see that it represents the number of states in delta.

Reply:
I am indifferent. num_of_states doesn't offend me, but if you have a better idea, let's rename it

Reply:
just that we did guess it both wrong with juraj